lib/stack_master/parameter_resolvers/latest_s3_object_version.rb

```diff
@@ -0,0 +1,22 @@
+module StackMaster
+  module ParameterResolvers
+    class LatestS3ObjectVersion < Resolver
+      array_resolver class_name: 'LatestS3ObjectVersions'
+
+      def initialize(config, stack_definition)
+        @config = config
+        @stack_definition = stack_definition
+        @s3_object_finder = S3ObjectFinder.new(@stack_definition.region)
+      end
+
+      def resolve(parameters)
+        parameters_map = @s3_object_finder.build_filters_from_hash(parameters)
```

Comment:
I would rather we solve the immediate use case and only take parameters we need, rather than solving for every use case by taking every possible parameter the S3 method can receive.
This would also help understand how this method can be used, right now it's not entirely clear what `parameters` needs to contain to get a result.
So can we limit the arguments `resolve()` takes to just those needed? (bucket and key?)